autowiring/auto_signal.h

```diff
@@ -507,12 +507,14 @@ namespace autowiring {
           state = SignalState::Free;
         } while (!m_state.compare_exchange_weak(state, SignalState::Asserting));
 
-        // Great.  If we got here it's because everyone got away and we're left holding the bag.
-        // We thought we were going to need to pend a callable signal but instead we're still going
-        // to need to invoke it ourselves.
         break;
       }
 
+      if (m_state == SignalState::Free) return;
```

Comment:
I'm not so sure this is the right answer anymore.  Comparing to `m_state` directly like this is pretty unsafe unless you're doing it in a loop, which we are not.